lib/chef/cookbook_version.rb

```diff
@@ -587,9 +587,7 @@ def yml_filenames_by_name(records)
     end
 
     def file_vendor
-      unless @file_vendor
-        @file_vendor = Chef::Cookbook::FileVendor.create_from_manifest(cookbook_manifest)
-      end
+      @file_vendor ||= Chef::Cookbook::FileVendor.create_from_manifest(cookbook_manifest)
       @file_vendor
```

Comment:
```suggestion
      @file_vendor ||= Chef::Cookbook::FileVendor.create_from_manifest(cookbook_manifest)
```

Reply:
the ||= should return the value either way...

Reply:
Yep. Added